### PR TITLE
284 [FEATURE] Project crew roles endpoint

### LINKED ISSUE #284: [FEATURE] Project crew roles endpoint
### Is there an existing issue or pull request for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

_No response_

### Describe the solution you'd like

v2 endpoint that lists project crew roles available to apply for and applications waiting for acception/rejection

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by cherry-john**
By submitting a PR for this repository you accept the contributor license agreement. 

### Description
Adds a crew role listing endpoint and a crew role application listing endpoint

### References
 Closes #284 

### Testing

- [x] Tested Locally

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in the documentation repo
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

**Comment by cherry-john** [comment]
Documented in adam-rms/website#5

File: html/admin/api/projects/crew/crewRoles/applicationList.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?php
+require_once __DIR__ . '/../../../apiHeadSecure.php';
+
+if (!$AUTH->instancePermissionCheck(123) or !isset($_POST['projectsVacantRoles_id'])) finish(false, ["code" => "PARAM-ERROR", "message"=> "No data for action"]);
```
Comment (Robert-Watts) [inline html/admin/api/projects/crew/crewRoles/applicationList.php:4]: If they do not have permission but have given data is it not a bit odd to say "no data"

File: html/admin/api/projects/crew/crewRoles/list.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?php
+require_once __DIR__ . '/../../../apiHeadSecure.php';
+
+if (!$AUTH->instancePermissionCheck(124)) die("404");
```
Comment (Robert-Watts) [inline html/admin/api/projects/crew/crewRoles/list.php:4]: is 404 the right response here? @Jbithell can correct me but I think the standard across the project is to use the "finish" function with an error message set

File: html/admin/api/projects/crew/crewRoles/applicationList.php (hunk 1 above)
Comment (Robert-Watts) [reply html/admin/api/projects/crew/crewRoles/applicationList.php:4]: like line 12

**Review by Robert-Watts — CHANGES_REQUESTED** [review]
Looks good see attached comments. My only other thing is, should this api route include issue #115 ? Just a thought?

**Comment by cherry-john** [comment]
> Looks good see attached comments. My only other thing is, should this api route include issue #115 ? Just a thought?

Probably not at this stage as that's adding a new feature

File: html/admin/api/projects/crew/crewRoles/list.php (hunk 2 above)
Comment (cherry-john) [reply html/admin/api/projects/crew/crewRoles/list.php:4]: AFAIK there's a mix of both, but generally 404s are returned when permission is the problem

Comment (cherry-john) [reply html/admin/api/projects/crew/crewRoles/list.php:4]: This follows the existing php file's action so I would keep it the same?

File: html/admin/api/projects/crew/crewRoles/applicationList.php (hunk 1 above)
Comment (cherry-john) [reply html/admin/api/projects/crew/crewRoles/applicationList.php:4]: Agreed, have changed

**Comment by Jbithell** [comment]
> Looks good see attached comments. My only other thing is, should this api route include issue #115 ? Just a thought?

No I don't think so - that's quite a complex change

**Comment by cherry-john** [comment]
@Robert-Watts could you re-review please!

**Review by Robert-Watts — APPROVED** [review]

**Review by Jbithell — APPROVED** [review]

**Pull request merged**